ci_scripts/utils/git_helpers.py

```diff
@@ -1,163 +1,415 @@
+
+    def __init__(self):
+        """Creates an instance of GitWrapper."""
+        self.repo = Repo(PROJECT_ROOT)
+        self.author = Actor("monty bot", "monty-bot@arm.com")
+
+    def _git_url_ssh_to_https(self, url: str) -> str:
+        """Changes repository URL to use authorisation token.
+
+        Converts the git url to use the GitHub token:
+        See https://github.blog/2012-09-21-easier-builds-and-deployments-using-git-over-https-and-oauth/
+
+        Returns:
+            new URL
+        """
+        path = url.split('github.com', 1)[1][1:].strip()
+        new = 'https://{GITHUB_TOKEN}:x-oauth-basic@github.com/%s' % path
+        logger.info('rewriting git url to: %s' % new)
+        return new.format(GITHUB_TOKEN=os.getenv(ENVVAR_GIT_TOKEN))
+
+    def configure_author(self):
+        """Sets the author."""
+        self.repo.config_writer().set_value("user", "name",
+                                            self.author.name).release()
+        self.repo.config_writer().set_value("user", "email",
+                                            self.author.email).release()
+
+    def checkout(self, branch) -> None:
+        """Checks out a branch.
+
+        Args:
+            branch: branch to check out
+
+        """
+        self.repo.git.checkout(branch)
+
+    def _add_one_file(self, path: str) -> None:
+        relative_path = os.path.relpath(path, start=PROJECT_ROOT)
+        if relative_path:
+            relative_path = relative_path.replace('\\', '/')
+            if os.path.isdir(relative_path):
+                relative_path = f'{relative_path}/*'
+        self.repo.git.add(relative_path)
+
+    def add(self, path) -> None:
+        """Adds a file or a list of files.
+
+        Args:
+            path: file path or list of file paths
+        """
+        if isinstance(path, list) or isinstance(path, set):
+            for element in path:
+                self.add(element)
+        else:
+            self._add_one_file(path)
+
+    def commit(self, message: str, **kwargs) -> None:
+        """Commits changes to the repository.
+
+        Args:
+            message: commit message
+            **kwargs: extra parameters
+        """
+        self.repo.index.commit(
+            message,
+            author=self.author,
+            **kwargs
+        )
+
+    def get_master_branch(self):
+        """Gets the `master` branch.
+
+        Returns:
+            corresponding branch
+        """
+        return self.get_branch(MASTER_BRANCH)
+
+    def get_beta_branch(self):
+        """Gets the `beta` branch.
+
+        Returns:
+            corresponding branch
+        """
+        return self.get_branch(BETA_BRANCH)
+
+    def is_release_branch(self, branch_name: str) -> bool:
+        """Checks whether the branch is a `release` branch or not.
+
+        Args:
+            branch_name: name of the branch
+
+        Returns: True if the branch is used for `release` code; False otherwise
```

Comment:
Desc. needs to go on a new line.

Reply:
Are these not enforced by the linter? 🤔

Reply:
Ah, maybe the branch hasn't been rebased to include the mypy stuff?

Reply:
it's been rebased and for some reason it was not picked up by flake8.

I blame flake 8 ;)

Reply:
Oh yes, sorry this is the docstrings not the typing. I'll have a look at why its not covered by Flake8.

Reply:
Ok so it looks like the linter doesn't pick it up because it isn't invalid - its just that the `Returns:` is no longer recognised as a section to be picked out specially in the documentation. 
Eg. on the same line it looks like this in the printed docs:
![Screenshot 2020-01-23 at 13 03 05](https://user-images.githubusercontent.com/32065424/72986791-e4464300-3de0-11ea-9cbf-3c1f2a9fd95a.png)
and if there is the line break:
![Screenshot 2020-01-23 at 13 02 45](https://user-images.githubusercontent.com/32065424/72986807-ec9e7e00-3de0-11ea-8d79-93b38220ce2b.png)

I don't think this can be enforced by Flake8, and its not the end of the world, but it's worth being aware of it if you can.

Reply:
Ah, that's good to know.